doc/src/guides/contributing/docstring.rst

```diff
@@ -156,7 +157,27 @@ is unnecessary, do not use it. Unnecessary sections and cluttered docstrings
 can make a function harder to understand. Aim for the minimal amount of
 information required to understand the function.
 
-1. Single-Sentence Summary
+1. Function Signature
+^^^^^^^^^^^^^^^^^^^^^
+
+The function signature is part of the Python source code but is almost always
+displayed along with the doc string. Function signatures should have
+informative argument names that follow Python variable name best practices,
+e.g. use ``coef_matrix`` instead of ``m``. It is also best to avoid the use of
```

Comment:
We might also consider starting to use positional-only arguments in cases where the argument name shouldn't be used as a keyword, which at least prevents the argument name from becoming part of the API.

Reply:
I rather think that this is a good example where type annotation can be better and make less verbose variables `coef: Matrix`, 
because we wouldn't need to make the names of the variable longer like [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation) if its semantics directly link to the type.